codegen/example/example_server.go

```diff
@@ -157,10 +157,22 @@ func main() {
 	{{- range .Server.Variables }}
 	{{ .VarName }}F = flag.String({{ printf "%q" .Name }}, {{ printf "%q" .DefaultValue }}, "{{ .Description }}{{ if .Values }} (valid values: {{ join .Values ", " }}){{ end }}")
 	{{- end }}
+		tlsCertPath   = flag.String("tls-cert", "", "TLS certificate file path (required in secure mode)")
```

Comment:
I can only call `flag.Parse` once. Otherwise I run into the following scenario: 
- starting the server in non-tls mode works without error
- starting the server by only passing `-secure` will yield the error `missing required argument argument -tls-cert` (as expected)
- starting the server by passing `-secure` and the tls paths will yield the error `flag provided but not defined: -tls-cert`

In other words, I must define this flag up front, or the flag package will throw an error.

Reply:
Sorry I should have been more clear: I meant that the code that define these flags should only be generated in the case where TLS is required (so there is a "https" scheme in one of the server URIs). The reason for this is that many setup have TLS terminated by a load balancer and thus it's actually pretty common to have server URIs that do not use https. In this common case we don't want to define command line flags that are not used. So there should be something like:
```go
{{ if hasTLS . }} // or something else...
tlsCertPath   = flag.String("tls-cert", "", "TLS certificate file path (required in secure mode)")
// ...
```
This means that code elsewhere that uses these flags also need to be protected with the same guard.